plugin/src/main/groovy/com/novoda/gradle/command/Input.groovy

```diff
@@ -1,80 +1,27 @@
 package com.novoda.gradle.command
 
+import org.gradle.api.Action
 import org.gradle.api.tasks.TaskAction
 
 class Input extends AdbTask {
 
-    Closure script
+    InputSpec script = new InputSpec()
 
-    void text(String value) {
-        input('text', "$value")
-    }
-
-    void tap(int x, int y) {
-        input('touchscreen', 'tap', x, y)
-    }
-
-    void swipe(int startX, int startY, int endX, int endY) {
-        input('touchscreen', 'swipe', startX, startY, endX, endY)
-    }
-
-    void key(int code) {
-        input('keyevent', code)
-    }
-    
-    void home() {
-        key 3
-    }
-    
-    void back() {
-        key 4
-    }
-
-    void up() {
-        key 19
-    }
-
-    void down() {
-        key 20
-    }
-    
-    void left() {
-        key 21
-    }
-
-    void right() {
-        key 22
-    }
-
-    void power() {
-        key 26
-    }
-    
-    void clear() {
-        key 28
-    }
-
-    void tab() {
-        key 61
-    }
-    
-    void enter() {
-        key 66
-    }
-
-    void unlock() {
-        key 82
-    }
-
-    private input(... values) {
-        def command = ["shell", "input"]
-        command.addAll(values)
-        runCommand(command)
+    @Deprecated
```

Comment:
I don't want to deprecate the field `script` because now it is a proper input that I use. It is now used when the task is created in the plugin. 

Before we had `Closure script`. With gradle magic, when the user creates a new task with type Input, they could use `script` like below (which looks like a method call)

This method is now introduced to support that the old behavior. 

```
task autoLogin(type: com.novoda.gradle.command.Input) {
     script {
    }
}
```

I didn't want to keep all the methods because that would cause a lot of duplication. I know that it is not ideal but with this we are 100% backward compatible. 

I will look into `@Delegate`. I guess with that I can just remove this method. I would have to remove deprecation but it is not really super visible anyways.